packages/user_packages/iotcc_mqtt_dev.py

```diff
@@ -0,0 +1,186 @@
+no_of_edge_system_in_thousands = 1
+# Number of Retries for Connection and Registrations
+no_of_retries_for_connection = 5
+# Retry delay Min Value in seconds
+delay_retries_min = 600
+# Retry delay Max Value in seconds
+delay_retries_max = 1800
+
+# Lambda Function Multiplier uses the above settings for calculating retry and delay logic
+lfm = lambda x: x * no_of_edge_system_in_thousands
+retry_attempts = lfm(no_of_retries_for_connection)
+delay_retries = random.randint(lfm(delay_retries_min), lfm(delay_retries_max))
+
+
+# ---------------------------------------------------------------------------
+# This is a sample application package to publish sample device stats to
+# IoTCC using MQTT protocol as DCC Comms
+# User defined methods
+
+def dev_metric():
+    """
+    This UDM randomly return values in between 0 to 999 in order to simulate device metric
+    Use case specific device metric collection logic should be provided by user
+    :return:
+    """
+    return randint(0, 999)
+
+
+class PackageClass(LiotaPackage):
+    def run(self, registry):
+        """
+        The execution function of a liota package.
+        Acquires "iotcc_mqtt" and "iotcc_mqtt_edge_system" from registry then register five devices
+        and publishes device metrics to the DCC
+        :param registry: the instance of ResourceRegistryPerPackage of the package
+        :return:
+        """
+        from liota.entities.devices.simulated_device import SimulatedDevice
+        from liota.entities.metrics.metric import Metric
+        import copy
+
+        # Acquire resources from registry
+        self.iotcc = registry.get("iotcc_mqtt")
+        # Creating a copy of edge_system object to keep original object "clean"
+        self.iotcc_edge_system = copy.copy(registry.get("iotcc_mqtt_edge_system"))
+
+        self.reg_devices = []
+        num_devices = 5
+
+        for i in range(0, num_devices):
+            try:
+                # Register device
+                device = SimulatedDevice(socket.gethostname() + "-ChildDev" + str(i))
+                log.info("Registration Started for Device {0}".format(device.name))
+                # Device Registration attempts
+                reg_attempts = 0
+                # Started Device Registration attempts
+                while reg_attempts <= retry_attempts:
+                    try:
+                        reg_dev = self.iotcc.register(device)
+                        break
+                    except Exception, e:
+                        if reg_attempts == retry_attempts:
+                            raise
+                        reg_attempts += 1
+                        log.error(
+                            '{0} Device Registration Failed with following error - {1}'.format(device.name, str(e)))
+                        log.info('{0} Device Registration: Attempt: {1}'.format(device.name, str(reg_attempts)))
```

Comment:
Doesn't have Trying word. Gateway registration messages had the word Trying.
This may help to find out similar logs with attempts.

Reply:
Added